alf/environments/igibson/suite_igibson.py

```diff
@@ -0,0 +1,384 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import gym
+import numpy as np
+import pybullet as p
+
+from alf.environments import suite_gym
+from alf.environments.gym_wrappers import BaseObservationWrapper, ImageChannelFirst, FrameGrayScale
+import alf
+
+from igibson.objects.visual_marker import VisualMarker
+from alf.environments.igibson.igibson_tasks import VisualPointNavFixedTask, \
+    VisualPointNavRandomTask, VisualObjectNavTask
+from alf.environments.igibson.igibson_env import iGibsonCustomEnv
+
+
+@alf.configurable()
+class FlattenWrapper(BaseObservationWrapper):
+    """Flatten selected obversation fields of the environment"""
+
+    def __init__(self, env, fields=None):
+        super().__init__(env, fields=fields)
+
+    def transform_space(self, observation_space):
+        if isinstance(observation_space, gym.spaces.Box):
+            if self._need_flatten(observation_space.shape):
+                low = observation_space.low
+                high = observation_space.high
+                if np.isscalar(low) and np.isscalar(high):
+                    shape = observation_space.shape[:-1]
+                    return gym.spaces.Box(
+                        low=low,
+                        high=high,
+                        shape=shape,
+                        dtype=observation_space.dtype)
+                else:
+                    low = self._flatten(observation_space.low, flatten=True)
+                    high = self._flatten(observation_space.high, flatten=True)
+                    return gym.spaces.Box(
+                        low=low, high=high, dtype=observation_space.dtype)
+        else:
+            raise TypeError('Unsupported observation space type')
+        return observation_space
+
+    def transform_observation(self, observation):
+        flatten = self._need_flatten(observation.shape)
+        return self._flatten(observation, flatten)
+
+    def _need_flatten(self, shape):
+        if len(shape) == 2:
+            return True
+        return False
+
+    def _flatten(self, np_array, flatten=False):
+        """
+        Note that the extra copy() after np.transpose is crucial to pickle dump speed
+        when called by subprocesses. An explanation of the non-contiguous memory caused
+        by numpy tranpose can be found in the following:
+
+        https://stackoverflow.com/questions/26998223/
+        """
+
+        if flatten:
+            np_array = np_array.flatten()
+        return np_array.copy()
+
+
+@alf.configurable()
+class StackRGBDWrapper(gym.ObservationWrapper):
```

Comment:
Probably we don't need this wrapper. We can instead define a network preprocessor ``NestConcat(dim=1)`` to concat two input tensors along the channels dim (assuming it's dim 1).